### PR TITLE
fix build for ghc 9.8.1 (#386)

### PR DESCRIPTION
**Pull request by arguri**
Before submitting your PR, check that you've:

- [x] Bumped the version number.
- [x] Ran `stylish-haskell` and otherwise adhered to the [style guide](https://github.com/bitemyapp/esqueleto/blob/master/style-guide.yaml).

After submitting your PR:

- [ ] Update the Changelog.md file with a link to your PR.
- [ ] Check that CI passes (or if it fails, for reasons unrelated to your change, like CI timeouts).

### CODE REVIEW

**Comment by Vekhir** [comment]
Don't you think that the changelog entry belongs under it's own header? Otherwise looks fine to me.

@parsonsmatt Do you have any suggestions for this PR adding support for GHC 9.8.1 to `esqueleto` before being merged?

File: changelog.md [hunk 1]
```diff
@@ -4,6 +4,9 @@
     - [#378](https://github.com/bitemyapp/esqueleto/pull/378)
         - `ToMaybe` instances are now derived for records so you can now left
           join them in queries
+- @arguri
+    - [#387](https://github.com/bitemyapp/esqueleto/pull/387)
+        - Fix build for ghc 9.8.1 / template-haskell 2.18
```
Comment (parsonsmatt) [inline changelog.md:9]: Please move this into it's own section of the changelog for this version.

File: src/Database/Esqueleto/Record.hs [hunk 2]
```diff
@@ -214,7 +214,9 @@ data RecordInfo = RecordInfo
     -- cause problems, but it's easy to pass around so might as well.
     constraints :: Cxt
   , -- | The original record's type-variable-binders.
-#if MIN_VERSION_template_haskell(2,17,0)
+#if MIN_VERSION_template_haskell(2,21,0)
+    typeVarBinders :: [TyVarBndr BndrVis]
+#elif MIN_VERSION_template_haskell(2,17,0)
```
Comment (parsonsmatt) [inline src/Database/Esqueleto/Record.hs:219]: Do you mind adding GHC 9.8 to the CI test to ensure this works?

**Review by parsonsmatt — CHANGES_REQUESTED** [review]
Thanks for the PR! There's two items left if you wouldn't mind.

File: changelog.md (hunk 1 above)
Comment (arguri) [reply changelog.md]: I hope I got that numbering right.

File: .github/workflows/haskell.yml [hunk 3]
```diff
@@ -33,7 +33,7 @@ jobs:
     strategy:
       matrix:
         cabal: ["3.8.1.0"]
-        ghc: ["8.6.5", "8.8.4", "8.10.4", "9.0.2", "9.2.2", "9.4.5", "9.6.2"]
+        ghc: ["8.6.5", "8.8.4", "8.10.4", "9.0.2", "9.2.2", "9.4.5", "9.6.2", "9.8.1"]
```
Comment (parsonsmatt) [inline .github/workflows/haskell.yml:36]: CI is failing due to this issue: https://github.com/haskellari/postgresql-libpq/issues/55

We may be able to fix this by adding an `allow-newer` flag to a `cabal.project`:

```
allow-newer:
  postgresql-libpq:Cabal
```

File: .github/workflows/haskell.yml [hunk 4]
```diff
@@ -33,7 +33,7 @@ jobs:
     strategy:
       matrix:
         cabal: ["3.8.1.0"]
```
Comment (parsonsmatt) [inline .github/workflows/haskell.yml:35]:
```suggestion
        cabal: ["3.10.2.1"]
```

**Review by parsonsmatt — APPROVED** [review]
Going to merge the suggestions to fix CI and hopefully get this in. Tested locally and it works great! Thank you!

File: .github/workflows/haskell.yml (hunk 3 above)
Comment (parsonsmatt) [reply .github/workflows/haskell.yml:36]: Nope, this was the *Cabal* version that needed to change 😅

**Comment by parsonsmatt** [comment]
Thank you!

**Pull request merged**